src/frontend/running.rs

```diff
@@ -342,7 +355,29 @@ impl RunningView {
                                 - self.farmer_state.initial_reward_address_balance;
                             self.farmer_state.initial_reward_address_balance =
                                 imported_block.reward_address_balance - previous_diff;
+                        } else {
+                            // only move the progress bar if it was stopped & the node is synced ofcourse
+                            if !self.reward_eta_progress_bar_moving {
+                                self.reward_eta_progress_bar_moving =
+                                    !self.reward_eta_progress_bar_moving;
+
+                                let total_space_pledged = total_space_pledged(
+                                    current_solution_range,
+                                    self.slot_probability,
+                                    max_pieces_in_sector,
+                                );
+
+                                let eta = calculate_expected_reward_duration_from_now(
+                                    total_space_pledged,
+                                    self.space_pledged,
+                                    self.last_reward_timestamp,
+                                )
+                                .map_err(|_| anyhow!("Failed to calculate ETA for reward payment"))
+                                .unwrap_or(0);
```

Comment:
Failed why? That shouldn't really be the case with a few numeric operations, also this particular way of handling it very odd: original error is ignored (so good luck figuring out why it failed), but also error is immediately ignored in `.unwrap_or(0)`, which makes error mapping pointless to begin with.

IMHO estimation function should return a meaningful result rather than error that is assumed to mean we're getting reward immediately.

FYI there is https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.unwrap_or_default